### DISCUSSION TITLE
Is it supported to limit the range of query results based on radius? and how？

### DISCUSSION THREAD

**Discussion by Wenorm** [milvus-io/milvus#4530] [Q&A and General discussion] [answered]
I read the documentation and source code, but I didn't find a way to set the radius to limit the number of returned results. Only found a way to set the number of returned results through the topk parameter.

**Comment by sahuang** [comment, marked as answer]
Currently, we only support returning `topK` results. Can you describe your scenario where a radius needs to be set? For example, if you set it to [10, 30], what should be returned?

If you are talking about range search, this is in our plan for future implementations.

**Reply by Wenorm** [reply]
I use milvus to retrieve similar texts. I use cosine similarity as the scoring criterion. So I want to set a value of cosine similarity to limit the number of returns, such as setting to only return vectors with cosine similarity greater than 0.5.

**Reply by sahuang** [reply]
I understand your point, this is range search which we haven't supported yet.
